src/init-nmos.js

```diff
@@ -53,7 +53,9 @@ function getPrioritised (representations, priority, version, protocol) {
             representation.protocol === protocol
       })[0]
     if (representation) {
-      if (representation.address.indexOf(':') > -1) {
+      if (protocol === 'https') {
+        url = `${protocol}://${representation.hostname}:${representation.port}`
+      } else if (representation.address.indexOf(':') > -1) {
         url = `${protocol}://[${representation.address}]:${representation.port}`
```

Comment:
I'll be honest I have no idea what this line is doing so if you can't either I'd take it out. Equally, if there is a reason for it then you'd need to apply the same logic to https mode i.e. 
```
url = `${protocol}://[${representation.hostname}]:${representation.port}`
```

Reply:
As discussed, the square brackets are for IPv6 addresses only